docs/manual/config/lazy.rst

```diff
@@ -109,6 +109,23 @@ Window functions
       - Put the focused window to/from floating mode
     * - ``lazy.window.toggle_fullscreen()``
       - Put the focused window to/from fullscreen mode
+    * - ``lazy.window.move_up()``
+      - Move the window above the next window in the stack.
+    * - ``lazy.window.move_down()``
+      - Move the window below the previous window in the stack.
+    * - ``lazy.window.move_to_top()``
+      - Move the window above all other windows with similar priority
+        (i.e. a "normal" window will not be moved above a ``kept_above`` window).
+    * - ``lazy.window.move_to_bottom()``
+      - Move the window below all other windows with similar priority
+        (i.e. a "normal" window will not be moved below a ``kept_below`` window).
+    * - ``lazy.window.keep_above()``
+      - Keep window above other windows.
+    * - ``lazy.window.keep_below()``
+      - Keep window below other windows.
+    * - ``lazy.window.bring_to_front()``
+      - Bring window above all other windows. Ignores ``kept_above`` priority.
```

Comment:
I think the question we need to consider is whether there'd ever be a scenario where someone wants to call `move_to_top` but still have it appear below things that are tagged as `kept_above`. If there isn't then we don't need the `respectful` arg.

Presumably, if you're moving to top you want to see the window so you want to ignore the `kept_above` tag for this purpose.

However, I think this causes a new issue: as the layer priorities have not been respected, you've now got a window that's out of place according to the spec. When you open a new window, it's going to open above the one with the same layer priority which means it will also open above windows that are `kept_above`. I guess we could solve that by adding additional logic which looks for `kept_above` windows and ensures new windows open below them.

Thoughts?

Reply:
Isn't the point of `kept_above` that you wouldn't want windows to get above it? A primary use-case I'm thinking of is some kind of monitor or video playing in the corner of the screen which you'd always want visible, while you perform otherwise normal operations on your windows including raising them.

Ultimately it's pretty difficult to guess what people would want! I'd maybe err on the side of not having the respectful flag, implementing whichever behaviour is easier (respecting or not), and then seeing what people want after that.

Reply:
Behaviour is as follows:
- `bring_to_top` puts window on top of everything
- `move_to_top` puts window at top of current layer
- passing `force=True` will remove `kept_below` flag if applied